src/bech32.cpp

```diff
@@ -396,23 +396,28 @@ DecodeResult Decode(const std::string& str) {
 }
 
 /** Find index of an incorrect character in a Bech32 string. */
-std::string LocateErrors(const std::string& str, std::vector<int>& error_locations) {
+std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str) {
```

Comment:
The docstring reads odd in doxygen:

"Find index of an incorrect character in a Bech32 string. Return the positions of errors in a Bech32 string. "

See https://doxygen.bitcoincore.org/namespacebech32.html#ad3d7336317a6e6d359b54b2f60b53acc

Also, `std::make_pair` is not needed with C++11/17

```diff
diff --git a/src/bech32.cpp b/src/bech32.cpp
index 3cda1dfff5..abc617467d 100644
--- a/src/bech32.cpp
+++ b/src/bech32.cpp
@@ -111,7 +111,7 @@ constexpr std::pair<std::array<int16_t, 1023>, std::array<int16_t, 1024>> Genera
         GF1024_LOG[v] = i;
     }
 
-    return std::make_pair(GF1024_EXP, GF1024_LOG);
+    return {GF1024_EXP, GF1024_LOG};
 }
 
 constexpr auto tables = GenerateGFTables();
@@ -395,27 +395,27 @@ DecodeResult Decode(const std::string& str) {
     return {result, std::move(hrp), data(values.begin(), values.end() - 6)};
 }
 
-/** Find index of an incorrect character in a Bech32 string. */
-std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str) {
+std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str)
+{
     std::vector<int> error_locations{};
 
     if (str.size() > 90) {
         error_locations.resize(str.size() - 90);
         std::iota(error_locations.begin(), error_locations.end(), 90);
-        return std::make_pair("Bech32 string too long", std::move(error_locations));
+        return {"Bech32 string too long", std::move(error_locations)};
     }
 
     if (!CheckCharacters(str, error_locations)){
-        return std::make_pair("Invalid character or mixed case", std::move(error_locations));
+        return {"Invalid character or mixed case", std::move(error_locations)};
     }
 
     size_t pos = str.rfind('1');
     if (pos == str.npos) {
-        return std::make_pair("Missing separator", std::vector<int>{});
+        return {"Missing separator", std::vector<int>{}};
     }
     if (pos == 0 || pos + 7 > str.size()) {
         error_locations.push_back(pos);
-        return std::make_pair("Invalid separator position", std::move(error_locations));
+        return {"Invalid separator position", std::move(error_locations)};
     }
 
     std::string hrp;
@@ -430,7 +430,7 @@ std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str) {
         int8_t rev = CHARSET_REV[c];
         if (rev == -1) {
             error_locations.push_back(i);
-            return std::make_pair("Invalid Base 32 character", std::move(error_locations));
+            return {"Invalid Base 32 character", std::move(error_locations)};
         }
         values[i - pos - 1] = rev;
     }
@@ -550,7 +550,7 @@ std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str) {
             }
         } else {
             // No errors
-            return std::make_pair("", std::vector<int>{});
+            return {"", std::vector<int>{}};
         }
 
         if (error_locations.empty() || (!possible_errors.empty() && possible_errors.size() < error_locations.size())) {
@@ -562,7 +562,7 @@ std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str) {
                               : error_encoding == Encoding::BECH32 ? "Invalid Bech32 checksum"
                               : "Invalid checksum";
 
-    return std::make_pair(error_message, std::move(error_locations));
+    return {error_message, std::move(error_locations)};
 }
 
 } // namespace bech32
diff --git a/src/bech32.h b/src/bech32.h
index 5e89e6efda..707299f19e 100644
--- a/src/bech32.h
+++ b/src/bech32.h
@@ -45,7 +45,7 @@ struct DecodeResult
 /** Decode a Bech32 or Bech32m string. */
 DecodeResult Decode(const std::string& str);
 
-/** Return the positions of errors in a Bech32 string. */
+/** Return the positions of errors in a Bech32 string. (Empty string if no errors can be found). */
 std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str);
 
 } // namespace bech32